File: 2585-delete-greatest-value-in-each-row/delete-greatest-value-in-each-row.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var deleteGreatestValue = function(grid) {
    for(let r of grid){
        r.sort((a,b)=>a-b);
    }
    let sum=0;
    for(var i=0;i<grid[0].length;i++){
        var max=0;
        for(var j=0;j<grid.length;j++){
            max=Math.max(max,grid[j][i]);
        }
        sum+=max;
    }
    return sum;
};

File: 2585-delete-greatest-value-in-each-row/delete-greatest-value-in-each-row.py
class Solution(object):
    def deleteGreatestValue(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        for row in grid:
            row.sort()
        sm=0
        for i in range(len(grid[0])):
            mx=grid[0][i]
            for j in range(len(grid)):
                mx=max(mx,grid[j][i])
            sm+=mx
        return sm